srml/balances/src/lib.rs

```diff
@@ -14,13 +14,149 @@
+//! - Getting and setting free balances
+//! - Retrieving total, reserved, and unreserved balances
+//! - Repatriating a reserved balance to a beneficiary account that exists
+//! - Transferring a balance between accounts (when not reserved)
+//! - Slashing an account balance
+//! - Account creation and removal
+//! - Lookup of an index to reclaim an account
+//! - Managing total issuance
+//! - Setting and managing locks
+//!
+//! ### Terminology
+//!
+//! - **Existential Deposit:** The minimum balance required to create or keep an account open. This prevents
+//! "dust accounts" from filling storage.
+//! - **Total Issuance:** The total amount of units in existence in a system.
+//! - **Reaping an account:** The act of removing an account by resetting its nonce. Happens after its balance is set
+//! to zero.
+//! - **Free Balance:** The liquid or spendable balance. The free balance is the only balance that matters for most
+//! operations. When this balance falls below the existential deposit, the account is removed.
+//! - **Reserved Balance:** Reserved balance still belongs to the account holder, but is suspended. Reserved balance
+//! can still be slashed, but only after all of free balance has been slashed. If the reserved balance falls below the
+//! existential deposit then it will be deleted.
+//! - **Imbalance:** A condition when some funds were created or deducted without equal and opposite accounting. Functions
+//! that result in an imbalance will return an object of the `Imbalance` trait that must be handled.
+//! - **Locks:** A freeze on an account's balance until a specified block number.
+//!
+//! ### Implementations
+//!
+//! The balances module provides implementations for the following traits. If these traits provide the functionality
+//! that you need, then you can avoid coupling with the balances module.
+//!
+//! - [`Currency`](https://crates.parity.io/srml_support/traits/trait.Currency.html): Functions for dealing with a
+//! fungible assets system.
+//! - [`LockableCurrency`](https://crates.parity.io/srml_support/traits/trait.LockableCurrency.html): Functions for
+//! dealing with accounts that allow liquidity restrictions.
+//! - [`Imbalance`]: Functions for handling imbalances between total issuance in the system and account balances.
+//! Must be used when a function creates new funds (e.g. a reward) or destroys some funds (e.g. a system fee).
+//! - [`MakePayent`]: Simple trait designed for hooking into a transaction payment. It operates over a single generic
+//! `AccountId` type.
+//! - [`IsDeadAccount`](https://crates.parity.io/srml_system/trait.IsDeadAccount.html): Determiner to say whether a
+//! given account is unused.
+//!
+//! Example of using the `Currency` trait from the treasury module:
+//!
+//! ```rust,ignore
+//! pub trait Trait: system::Trait {
+//! 	/// The staking balance.
+//! 	type Currency: ArithmeticType + Currency<Self::AccountId, Balance=<<Self as Trait>::Currency as ArithmeticType>::Type>;
```

Comment:
Indeed this was an important example when we used ArithmeticType coupling to constrain the balance but now we deciding to simplify this by making the arithmeticType constraint of balance in Currency itself, so the example is outdated, less important but still relevant to me.
```suggestion
//! 	type Currency: Currency<Self::AccountId>;
```